packages/teleterm/src/ui/StatusBar/ShareFeedback/ShareFeedback.test.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { fireEvent, render } from 'design/utils/testing';
+import { ShareFeedback } from './ShareFeedback';
+import { MockAppContextProvider } from 'teleterm/ui/fixtures/MockAppContextProvider';
+import { ClustersService } from 'teleterm/ui/services/clusters';
+import AppContext from 'teleterm/ui/appContext';
+import { WorkspacesService } from 'teleterm/ui/services/workspacesService';
+import { Cluster } from 'teleterm/services/tshd/v1/cluster_pb';
+
+jest.mock('teleterm/ui/services/clusters');
+jest.mock('teleterm/ui/services/workspacesService');
+
+function getTestSetup() {
+  const clustersService = new ClustersService(undefined, undefined);
+  const workspacesService = new WorkspacesService(
+    undefined,
+    undefined,
+    undefined,
+    undefined
+  );
+
+  jest
+    .spyOn(ClustersService.prototype, 'findCluster')
+    .mockImplementation(clusterUri => {
+      if (clusterUri === '/clusters/localhost')
+        return {
+          loggedInUser: { name: 'alice' },
+        } as Cluster.AsObject;
+
+      if (clusterUri === '/clusters/production') {
+        return {
+          loggedInUser: {
+            name: 'bob@prod.com',
+          },
+        } as Cluster.AsObject;
+      }
```

Comment:
Could you do this within the scope of the test itself? From my experience, a test setup like this (a single mock which behavior depends on a value from tests) gets progressively harder to understand and maintain as the size of the test suite grows.

Instead of having to look just at the test body itself to determine what the behavior will be, you now also have to inspect functions defined in other parts of the test suite.